csharp/src/Ice/Object.cs

```diff
@@ -67,118 +75,72 @@ public string[] ice_ids(Current current)
-        inS.setResult(ostr);
-        return null;
+        InputStream istr = request.inputStream;
+        istr.skipEmptyEncapsulation();
+        string ret = obj.ice_id(request.current);
+        return new(request.current.createOutgoingResponse(ret, static (ostr, ret) => ostr.writeString(ret)));
     }
+}
 
+/// <summary>
+/// Base class for all Slice classes.
+/// </summary>
+public abstract class ObjectImpl : Object
+{
     /// <summary>
     /// Returns the Slice type ID of the most-derived interface supported by this object.
     /// </summary>
     /// <param name="current">The Current object for the dispatch.</param>
     /// <returns>The return value is always ::Ice::Object.</returns>
     public virtual string ice_id(Current current) => ice_staticId();
 
-    [EditorBrowsable(EditorBrowsableState.Never)]
-    public static Task<OutputStream>? iceD_ice_id(Object obj, Ice.Internal.Incoming inS, Current current)
-    {
-        inS.readEmptyParams();
-        var ret = obj.ice_id(current);
-        var ostr = inS.startWriteParams();
-        ostr.writeString(ret);
-        inS.endWriteParams(ostr);
-        inS.setResult(ostr);
-        return null;
-    }
-
     /// <summary>
     /// Returns the Slice type ID of the interface supported by this object.
     /// </summary>
     /// <returns>The return value is always ::Ice::Object.</returns>
     public static string ice_staticId() => "::Ice::Object";
 
-    private static readonly string[] _all = new string[]
-    {
-        "ice_id", "ice_ids", "ice_isA", "ice_ping"
-    };
-
-    [EditorBrowsable(EditorBrowsableState.Never)]
-    public virtual Task<OutputStream>? iceDispatch(Ice.Internal.Incoming inc, Current current)
-    {
-        int pos = Array.BinarySearch(_all, current.operation);
-        if (pos < 0)
+    public virtual ValueTask<OutgoingResponse> dispatchAsync(IncomingRequest request) =>
+        request.current.operation switch
         {
-            throw new OperationNotExistException(current.id, current.facet, current.operation);
-        }
-
-        switch (pos)
-        {
-            case 0:
-            {
-                return iceD_ice_id(this, inc, current);
-            }
-            case 1:
-            {
-                return iceD_ice_ids(this, inc, current);
-            }
-            case 2:
-            {
-                return iceD_ice_isA(this, inc, current);
-            }
-            case 3:
-            {
-                return iceD_ice_ping(this, inc, current);
-            }
-        }
-
-        Debug.Assert(false);
-        throw new OperationNotExistException(current.id, current.facet, current.operation);
-    }
+            "ice_id" => Object.iceD_ice_idAsync(this, request),
+            "ice_ids" => Object.iceD_ice_idsAsync(this, request),
+            "ice_isA" => Object.iceD_ice_isAAsync(this, request),
+            "ice_ping" => Object.iceD_ice_pingAsync(this, request),
+            _ => throw new OperationNotExistException()
```

Comment:
Do we catch this and  "fill it in" somewhere else

Reply:
Yes, when we create the response from an exception.